cli/cli.go

```diff
@@ -448,12 +450,29 @@ func (runOptions *runOptionsType) handleCLIOptions(ctx *cli.Context) error {
 func (runOptions *runOptionsType) setupCLIHandler(ctx *cli.Context) error {
 	if len(ctx.Args()) > 0 {
 		return errors.Errorf(
-			errMsgAmbiguousArgumentsGiven,
+			errMsgAmbiguousArgumentsGivenF,
 			ctx.Args().First())
 	}
 	if !ctx.GlobalIsSet("log-level") {
 		log.SetLevel(log.WarnLevel)
 	}
+
+	// Check implicit flag dependencies
+	if ctx.IsSet("server-url") || ctx.IsSet("server-ip") {
+		if ctx.IsSet("server-url") && ctx.IsSet("server-ip") {
+			return errors.Errorf(errMsgConflictingArgumentsF,
+				"server-url", "server-ip")
+		} else if ctx.IsSet("server-ip") {
+			ctx.Set("demo", "true")
+		}
+		ctx.Set("mender-professional", "false")
+	}
```

Comment:
I still have this strange case. Setting `server-ip` and any (or all) of the interval flags (i.e. --update-poll 123) will then prompt me for serverURL.

Reply:
Yeah, I can see the changes due to setting --update-poll takes precedence over `--server-ip` what is the expected behavior in this case? Should I raise a conflicting flags error? should we treat `--server-ip` as `--server-url`?